lib/mimemagic.rb

```diff
@@ -62,16 +62,39 @@ def comment
   end
 
   # Lookup mime type by file extension
-  def self.by_extension(ext)
+  # Option keys:
+  # * <i>:multiple</i>: Return an array with all MIME types found. Instead only the first match is returned.
+  def self.by_extension(ext, options = {})
     ext = ext.to_s.downcase
     mime = ext[0..0] == '.' ? EXTENSIONS[ext[1..-1]] : EXTENSIONS[ext]
-    mime && new(mime)
+    case mime
+    when Array
+      mime = mime.map { |e| e && new(e) }
+      options[:multiple] ? mime : mime.first
+    when String
+      options[:multiple] ? [mime] : mime && new(mime)
+    end
   end
```

Comment:
You can write this much better

~~~ ruby
ext = ext[1..-1] if ext [0..0]
mime = [*EXTENSIONS[ext]]
options[:multiple] ? mime : (mime.first && new(mime.first))
~~~